### PR TITLE
core(driver): wait for critical network requests

### LINKED ISSUE #10810: Lighthouse doesn't wait long enough when throttlingMethod=simulate
#### Provide the steps to reproduce
`lighthouse --view https://develop.pwa-venia.com/venia-bottoms.html`

#### What is the current behavior?
Currently lighthouse doesn't wait long enough to detect the Cumulative Layout Shift for some pages.

Schermafbeelding 2020-05-19 om 15 07 40

#### What is the expected behavior?
I'd expect to report a lower score for the Cumulative Layout Shift on the page, because the page will actually load later.

#### Environment Information
* Affected Channels: cli
* Lighthouse version: 6

### LINKED ISSUE #11180: Lighthouse doesn't wait long enough for page to load (NO_LCP)
#### Steps to reproduce
1. Run `lighthouse https://clientes-bancocaminos.grupocaminos.es`
2. Open report
  
#### What is the current behavior?

On CLI I get: 
`largest-contentful-paint:warn Caught exception: NO_LCP`

On DevTools I get:
`Largest Contentful Paint Error!`
`Something went wrong with recording the trace over your page load. Please run Lighthouse again. (NO_LCP)`

#### What is the expected behavior?
I don't expect any errors.

#### Environment Information
* Affected Channels: CLI, DevTools
* Lighthouse version: 6.1.1
* Chrome version: 84.0.4147.105

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Waits for all critical (high priority) network requests to finish before continuing to the CPU idle check. This solves the issue when the entire page is gated on a single XHR but we move on because we're already at network 2-quiet. This also addresses the problem where the LCP image is very large and added via script where we optimistically report the LCP too early.

The net effect here will be Lighthouse taking a little longer for each run and performance scores dipping as more resources and more CPU work is included in our observations. The net effect on popular sites I tried that don't face the failure case was relatively small. The smoketest and URLs in the linked issues illustrate the massive failure cases we're addressing.

- CNN, waits an additional ~1s, no observable effect on scores
- The Verge, waits an additional ~2s, no observable effect on scores
- ESPN, waits an additional ~1s, no observable effect on scores

**Related Issues/PRs**
fixes #11180 
fixes #10810 
fixes https://github.com/GoogleChrome/lighthouse/issues/11639

### CODE REVIEW

File: lighthouse-core/lib/i18n/locales/en-US.json [hunk 1]
```diff
@@ -2111,50 +2111,8 @@
   "node_modules/lighthouse-stack-packs/packs/magento.js | uses-webp-images": {
     "message": "Consider searching the [Magento Marketplace](https://marketplace.magento.com/catalogsearch/result/?q=webp) for a variety of third-party extensions to leverage newer image formats."
   },
-  "node_modules/lighthouse-stack-packs/packs/october.js | efficient-animated-content": {
```
Comment (connorjclark) [inline lighthouse-core/lib/i18n/locales/en-US.json:2114]: what happened here?

Comment (patrickhulce) [reply lighthouse-core/lib/i18n/locales/en-US.json:2114]: running update:sample-json without yarn install, reverting

File: lighthouse-cli/test/smokehouse/test-definitions/tricky-metrics/expectations.js [hunk 2]
```diff
@@ -44,6 +44,18 @@ module.exports = [
...
     },
   },
+  {
+    lhr: {
+      requestedUrl: 'http://localhost:10200/delayed-lcp.html',
+      finalUrl: 'http://localhost:10200/delayed-lcp.html',
+      audits: {
+        'largest-contentful-paint': {
+          // LCP is after the ~7s XHR and the ~7s image.
+          numericValue: '>13500',
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/tricky-metrics/expectations.js:54]: Why isn't this strictly >14s?

File: lighthouse-core/gather/driver/wait-for-condition.js [hunk 3]
```diff
@@ -112,10 +113,10 @@ function waitForFcp(session, pauseAfterFcpMs, maxWaitForFcpMs) {
  * `networkQuietThresholdMs` ms and a method to cancel internal network listeners/timeout.
  * @param {LH.Gatherer.FRProtocolSession} session
  * @param {NetworkRecorder} networkMonitor
```
Comment (connorjclark) [inline lighthouse-core/gather/driver/wait-for-condition.js:115]: any reason this isn't called `networkRecorder`?

Comment (patrickhulce) [reply lighthouse-core/gather/driver/wait-for-condition.js:115]: just an artifact of how it used to be in driver, though honestly `networkMonitor` does better express it's usage almost everywhere other than for devtoolsLog creation :)

File: lighthouse-core/test/gather/driver/wait-for-condition-test.js [hunk 4]
```diff
@@ -35,6 +35,23 @@ function createMockWaitForFn() {
...
+    const {promise, resolve, reject} = createDecomposedPromise();
+    calls.push({
+      arguments: args,
+      mockResolve: () => resolve(),
+      mockReject: () => reject(new Error('Rejected')),
+    });
+    return {promise, cancel: mockCancelFn};
+  });
+
+  return Object.assign(mockFn, {waitForCalls: calls});
```
Comment (connorjclark) [inline lighthouse-core/test/gather/driver/wait-for-condition-test.js:52]: 👀

**Review by connorjclark — APPROVED** [review]

File: lighthouse-core/gather/driver/wait-for-condition.js (hunk 3 above)
Comment (connorjclark) [reply lighthouse-core/gather/driver/wait-for-condition.js:115]: it always confused me b/c there is no network-monitor file. would be nice to rename one or the other

File: lighthouse-cli/test/smokehouse/test-definitions/tricky-metrics/expectations.js (hunk 2 above)
Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/test-definitions/tricky-metrics/expectations.js:54]: paranoia :) 

IIRC this habit of mine came from a time when our time origin was less reliable that caused flakes of up to a few 100 ms.  14s should be fine these days.
```suggestion
          numericValue: '>14000',
```

File: lighthouse-core/test/gather/driver/wait-for-condition-test.js (hunk 4 above)
Comment (patrickhulce) [reply lighthouse-core/test/gather/driver/wait-for-condition-test.js:52]: if these eyes are for preferring `{...` that won't work for prototype/class properties of the jest mock fn :)

Comment (connorjclark) [reply lighthouse-core/test/gather/driver/wait-for-condition-test.js:52]: ah ya I know. I was just eying the rare case where my blanket "use spread!" doesn't work :)

**Pull request merged**